src/client/caching/structures/guilds/Role.ts

```diff
@@ -0,0 +1,77 @@
+import { Structure } from '../base/Structure';
+
+import type { APIRoleData } from '@klasa/dapi-types';
+import type { Client } from '../../../Client';
+import { Permissions } from '../../../../util/bitfields/Permissions';
+
+/**
+ * @see https://discord.com/developers/docs/topics/permissions#role-object
+ */
+export class Role extends Structure {
+
+	/**
+	 * The role's ID.
+	 * @since 0.0.1
+	 */
+	public readonly id: string;
+
+	/**
+	 * The role's name.
+	 * @since 0.0.1
+	 */
+	public name!: string;
+
+	/**
+	 * The role's integer representation of hexadecimal color code.
+	 * @since 0.0.1
+	 */
+	public color!: number;
+
+	/**
+	 * Whether or not the role is pinned in the user listing.
+	 * @since 0.0.1
+	 */
+	public hoist!: boolean;
+
+	/**
+	 * The role's position.
+	 * @since 0.0.1
+	 */
+	public position!: number;
+
+	/**
+	 * The role's permissions.
+	 * @since 0.0.1
+	 */
+	public permissions!: Readonly<Permissions>;
+
+	/**
+	 * Whether or not the role is managed by an integration.
+	 * @since 0.0.1
+	 */
+	public readonly managed!: boolean;
```

Comment:
Why is this one readonly of all of them

Reply:
Is there the possibility of a role changing this state?

Reply:
That's a valid point, it probably can't change..Or at least I'd hope so >w>